lib/ecto/adapters/myxql/connection.ex

```diff
@@ -549,14 +549,14 @@ if Code.ensure_loaded?(MyXQL) do
     defp order_by(%{order_bys: order_bys} = query, sources) do
       [
         " ORDER BY "
-        | Enum.map_intersperse(order_bys, ", ", fn %QueryExpr{expr: expr} ->
+        | Enum.map_intersperse(order_bys, ", ", fn %ByExpr{expr: expr} ->
             Enum.map_intersperse(expr, ", ", &order_by_expr(&1, sources, query))
           end)
       ]
     end
 
     defp order_by_expr({dir, expr}, sources, query) do
-      str = expr(expr, sources, query)
+      str = expr({:top_level, expr}, sources, query)
```

Comment:
Instead of wrapping it in a tuple, have a separate function called `top_level_expr(expr, sources, query)`, that falls back to expr(...) if not a subquery.

Reply:
Yep, thats better :)